File: Data Structures & Algorithms/anagram-groups/submission-2.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        const res = {};

        for(let s of strs){
            const count = new Array(26).fill(0);
            // console.log(count);
            for(let c of s){
                count[c.charCodeAt(0) - 'a'.charCodeAt(0)] += 1;
            }
            const key = count.join(',');

            // putIfAbsent
            if(!res[key]){
                res[key] = [];
            }

            res[key].push(s);
            console.log(res);
        }
        return Object.values(res);
    }
}

File: Data Structures & Algorithms/count-paths/submission-2.java
public class Solution {


    public static void main(String[] args){
        uniquePaths(3, 22);
    }
    public static int uniquePaths(int m, int n) {
        if (m == 1 || n == 1) {
            return 1;
        }
        if (m < n) {
            int temp = m;
            m = n;
            n = temp;
        }

        long res = 1;
        int j = 1;
        for (int i = m; i < m + n - 1; i++) {
            res *= i;
            res /= j;
            j++;
        }

        return (int) res;
    }
}